lib/console.js

```diff
@@ -249,6 +265,98 @@ Console.prototype.groupEnd = function groupEnd() {
     this[kGroupIndent].slice(0, this[kGroupIndent].length - 2);
 };
 
+const keyKey = 'Key';
+const valuesKey = 'Values';
+const indexKey = '(index)';
+const iterKey = '(iteration index)';
+
+
+const isArray = (v) => ArrayIsArray(v) || isTypedArray(v) || isBuffer(v);
+const inspect = (v) => {
+  const opt = { depth: 0, maxArrayLength: 3 };
+  if (v !== null && typeof v === 'object' &&
+      !isArray(v) && ObjectKeys(v).length > 2)
+    opt.depth = -1;
+  return util.inspect(v, opt);
+};
+
+const getIndexArray = (n) => ArrayFrom({ length: n }, (_, i) => inspect(i));
+
+// https://console.spec.whatwg.org/#table
+Console.prototype.table = function(tabularData, properties) {
+  if (properties !== undefined && !ArrayIsArray(properties)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'properties',
+                               'array', properties);
+  }
+
+  if (tabularData == null ||
+      (typeof tabularData !== 'object' && typeof tabularData !== 'function') ||
+      (!isArray(tabularData) && ObjectKeys(tabularData) > 20))
```

Comment:
While looking into this deeper: limiting keys here is not necessary. It is necessary though for nested objects / functions and that is not done currently. It would also be good to add a test for this.